### PR TITLE
Docs: Fix typos and grammar

### PR DESCRIPTION
**Pull request by eriksjolund**
**What this PR does / why we need it**:

Fixes typos and grammar. Most of them were occurrences of `it's` that should be `its`.

**Which issue(s) this PR fixes**:

None

**Special notes for your reviewer**:

**Checklist**
- [x] Reviewed the [`CONTRIBUTING.md`](https://github.com/grafana/loki/blob/main/CONTRIBUTING.md) guide (**required**)
- [ ] Documentation added
- [ ] Tests updated
- [ ] `CHANGELOG.md` updated
- [ ] Changes that require user attention or interaction to upgrade are documented in `docs/sources/upgrading/_index.md`

### CODE REVIEW

File: docs/sources/installation/helm/reference.md [hunk 1]
```diff
@@ -1993,7 +1993,7 @@ false
 		<tr>
 			<td>migrate.fromDistributed.memberlistService</td>
 			<td>string</td>
-			<td>If migrating from a distributed service, provide the distributed deployment's memberlist service DNS so the new deployment can join it's ring.</td>
+			<td>If migrating from a distributed service, provide the distributed deployment's memberlist service DNS so the new deployment can join its ring.</td>
```
Comment (JStickler) [inline docs/sources/installation/helm/reference.md:1996]: I believe that this file is generated and the one you want to edit is loki/production/helm/loki/values.yaml.

**Review by JStickler — COMMENTED** [review]
[docs team] Thanks so much for fixing these typos!  Very much appreciated.

See comment on the Helm documentation.

**Comment by eriksjolund** [comment]
Should I remove the edits under _docs/sources/installation_?
Do you want me fix incorrect occurences of `it's` under _tools/_, _pkg/_ and _client/_?

For instance
```
$ grep -r "it's own" tools/ | wc -l         
       1
$ grep -r "it's own" pkg/ | wc -l  
       6
$ grep -r "it's own" clients/ | wc -l 
       1
$ 
```

If so I would need to do it over the weekend.

File: docs/sources/installation/helm/reference.md (hunk 1 above)
Comment (eriksjolund) [reply docs/sources/installation/helm/reference.md:1996]: Fixed

**Review by trevorwhitney — APPROVED** [review]
lgtm, thanks for the contribution!

**Pull request merged**